src/openms/source/ANALYSIS/OPENSWATH/MRMScoring.cpp

```diff
@@ -213,38 +185,26 @@ namespace OpenSwath
 
     void MRMScoring::initializeXCorrPrecursorCombinedMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& precursor_ids, const std::vector<String>& native_ids)
     {
-      std::vector<double> intensityi, intensityj;
-      std::vector<FeatureType> features;
-
-      for (std::size_t i = 0; i < precursor_ids.size(); i++)
+      std::vector<std::vector<double>> intensityi, intensityj;
```

Comment:
this method looks rather time consuming because it operates on a combined matrix (which is 4x the size).
Does this acutally show up as time consuming in the flame graph? (how much is is compared to the others?)

Could this be:
1) computed more efficiently by computing normalized vectors (what is only done very late in `Scoring::normalizedCrossCorrelation` currently) for both Precusor and ID data separately ?
 And only then combine the vectors into a big matrix?

2) compute not the full matrix but only the triangle + diagonal (half the work)?